examples/usages/context_menu.rs

```diff
@@ -0,0 +1,190 @@
+//! This example illustrates how to create a context menu that changes the clear color
+
+use bevy::{
+    color::palettes::basic,
+    ecs::{relationship::RelatedSpawner, spawn::SpawnWith},
+    prelude::*,
+};
+use std::fmt::Debug;
+
+#[derive(Event)]
+struct OpenContextMenu {
+    pos: Vec2,
+}
+
+#[derive(Event)]
+struct CloseContextMenus;
+
+#[derive(Component)]
+struct ContextMenu;
+
+#[derive(Component)]
+struct ContextMenuItem(Srgba);
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_observer(on_trigger_menu)
+        .add_observer(on_trigger_close_menus)
+        .add_observer(create_hover_observer::<Out>(basic::WHITE))
+        .add_observer(create_hover_observer::<Over>(basic::RED))
+        .run();
+}
+
+fn create_hover_observer<T: Debug + Clone + Reflect>(
+    color: Srgba,
+) -> impl FnMut(Trigger<Pointer<T>>, Query<&mut TextColor>, Query<&Children>) {
+    move |trigger: Trigger<Pointer<T>>,
+          mut query: Query<&mut TextColor>,
+          children: Query<&Children>| {
+        let Ok(children) = children.get(trigger.target()) else {
+            return;
+        };
```

Comment:
My curiosity talking if someone knows the answer, we saw that this was this trigger fires for the window ; is this worth a comment ?

And can we do better (as further work to improve API, I'd have expected a different `Pointer<WindowOut>` ?)?

Reply:
I believe we should stop propagating here to not enter in mentioned behaviour above:

```suggestion
        let Ok(children) = children.get(trigger.target()) else {
            return;
        };
        trigger.propagate(false);
```

Reply:
calling `propagate(false)` here does not help or change anything sadly

Reply:
I think it does:

With `propagate(false)`:
```
2025-05-18T11:13:26.024817Z  INFO context_menu: background: Some("Over")
2025-05-18T11:13:27.590187Z  INFO context_menu: background: Some("Out")
2025-05-18T11:13:27.590241Z  INFO context_menu: button: Some("Over")
2025-05-18T11:13:28.605832Z  INFO context_menu: button: Some("Out")
2025-05-18T11:13:28.605878Z  INFO context_menu: background: Some("Over")
```

Without:
```
2025-05-18T11:12:21.474795Z  INFO context_menu: background: Some("Over")
2025-05-18T11:12:21.474829Z  INFO context_menu: 0v0: Some("Over")
2025-05-18T11:12:22.257533Z  INFO context_menu: background: Some("Out")
2025-05-18T11:12:22.257576Z  INFO context_menu: 0v0: Some("Out")
2025-05-18T11:12:22.257598Z  INFO context_menu: button: Some("Over")
2025-05-18T11:12:22.257613Z  INFO context_menu: background: Some("Over")
2025-05-18T11:12:22.257623Z  INFO context_menu: 0v0: Some("Over")
2025-05-18T11:12:24.109703Z  INFO context_menu: button: Some("Out")
2025-05-18T11:12:24.109768Z  INFO context_menu: background: Some("Out")
2025-05-18T11:12:24.109780Z  INFO context_menu: 0v0: Some("Out")
2025-05-18T11:12:24.109803Z  INFO context_menu: background: Some("Over")
2025-05-18T11:12:24.109814Z  INFO context_menu: 0v0: Some("Over")
```

<details><summary>Diff</summary>
<p>

```diff
diff --git a/examples/usages/context_menu.rs b/examples/usages/context_menu.rs
index 9ab055d9b..f46b89abb 100644
--- a/examples/usages/context_menu.rs
+++ b/examples/usages/context_menu.rs
@@ -39,13 +39,20 @@ fn main() {
 // helper function to reduce code duplication when generating almost identical observers for the hover text color change effect
 fn create_hover_observer<T: Debug + Clone + Reflect>(
     color: Srgba,
-) -> impl FnMut(Trigger<Pointer<T>>, Query<&mut TextColor>, Query<&Children>) {
-    move |trigger: Trigger<Pointer<T>>,
+) -> impl FnMut(Trigger<Pointer<T>>, Query<&mut TextColor>, Query<&Children>, Query<NameOrEntity>) {
+    move |mut trigger: Trigger<Pointer<T>>,
           mut query: Query<&mut TextColor>,
-          children: Query<&Children>| {
+          children: Query<&Children>,
+          name: Query<NameOrEntity>| {
+        info!(
+            "{}: {:?}",
+            name.get(trigger.target()).unwrap(),
+            trigger.event().reflect_type_ident()
+        );
         let Ok(children) = children.get(trigger.event().target) else {
             return;
         };
+        trigger.propagate(false);
 
         // find the text among children and change its color
         for child in children.iter() {
```

</p>
</details> 

I'm not sure exactly what's the entity it's propagating to, but we don't need it.

Reply:
interesting. ok added it